m27q.py

```diff
@@ -3,14 +3,49 @@
 # Based on: https://gist.github.com/wadimw/4ac972d07ed1f3b6f22a101375ecac41
 
 
+import sys
+import typing as t
+from dataclasses import dataclass
+from time import sleep
 import usb.core
 import usb.util
-from time import sleep
-import typing as t
-import sys
+
+
+@dataclass
+class BasicProperty:
+    minimum: int
+    maximum: int
+    message_a: int
+    message_b: int = 0
+
+    def clamp(self, v: int):
+        return max(self.minimum, min(self.maximum, v))
+
+
+@dataclass
+class EnumProperty:
+    allowed: t.List[int]
+    message_a: int
+    message_b: int = 0
+
+    def clamp(self, v: int):
+        if v not in self.allowed:
+            raise Exception(f"Only allowed values: {self.allowed}")
+        return v
```

Comment:
I wonder if we could have an abstract base property class with the common attributes plus a method that returns the list expected by `set_osd`.